patches/tModLoader/Terraria/Social/Steam/WorkshopSocialModule.TML.cs

```diff
@@ -247,11 +250,32 @@ internal static HashSet<string> DetermineSupportedVersionsFromWorkshop(string re
 		return summary.Select(info => SocialBrowserModule.GetBrowserVersionNumber(info.tModVersion)).ToHashSet();
 	}
 
-	internal static LocalMod OpenModFile(string path)
+	// PR 4345 - We combine the hash data that is currently on workshop with the hash data from the updated publishing folder to ensure that when mods are updated it is backwards compatible
+	// It is backwards compatible while Steam spends up to an hour rolling out workshop item updates
+	/// <summary>
+	/// Gets the revised Developer Metadata fo usage with publishing a new mod or update to an existing mod.
+	/// Takes the folder path containing all .tmod files and the PublishFileID. A PublishFileID of zero is a new mod by convention.
+	/// </summary>
+	internal static DeveloperMetadata GetDeveloperMetadataForPublish(string folderPath, ulong publishId)
 	{
-		var sModFile = new TmodFile(path);
-		using (sModFile.Open())
-			return new LocalMod(ModLocation.Workshop, sModFile);
+		var newDeveloperMetadata = new DeveloperMetadata();
+
+		var currentHashes = ModOrganizer.GetModHashesFromFolder(folderPath);
+
+		// Publish ID == 0 is a new mod
+		if (publishId == 0) {
+			newDeveloperMetadata.modVersionHashes = currentHashes;
+			return newDeveloperMetadata;
+		}
+
+		var pubId = new ModPubId_t() { m_ModPubId = publishId.ToString() };
+		var oldDevMetadata = WorkshopBrowserModule.Instance.GetDeveloperMetadataFromModBrowser(pubId);
+
+		newDeveloperMetadata.modVersionHashes = currentHashes.Concat(oldDevMetadata.modVersionHashes.Except(currentHashes).ToList()).ToList();
+
+		newDeveloperMetadata.TrimDevMetadataForPublish();
+
+		return newDeveloperMetadata;
```

Comment:
This code will need maintenance if we ever add anything new to developer metadata that isn't recalculated each time in this function. I think it would be better if we just did something like

```cs
var developerMetadata = WorkshopBrowserModule.Instance.GetDeveloperMetadataFromModBrowser(pubId);
developerMetadata.modVersionHashes = currentHashes.Concat(developerMetadata.modVersionHashes.Except(currentHashes).ToList()).ToList();
developerMetadata .TrimDevMetadataForPublish();
return developerMetadata;
```

I think `GetDeveloperMetadataFromModBrowser` will return an empty metadata for 0 publishid, but if not, we can `: new()` it